--- ../src-base/minecraft/net/minecraft/tileentity/TileEntity.java
+++ ../src-work/minecraft/net/minecraft/tileentity/TileEntity.java
@@ -20,6 +20,7 @@
 import net.minecraftforge.fml.relauncher.SideOnly;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.bukkit.inventory.InventoryHolder;
 
 public abstract class TileEntity implements net.minecraftforge.common.capabilities.ICapabilitySerializable<NBTTagCompound>
 {
@@ -537,4 +538,12 @@
         addMapping(TileEntityEndGateway.class, "EndGateway");
         addMapping(TileEntityCommandBlock.class, "Control");
     }
+    // CraftBukkit start - add method
+    public InventoryHolder getOwner() {
+    	if (worldObj == null) return null;
+    	org.bukkit.block.BlockState state = worldObj.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ()).getState();
+    	if (state instanceof InventoryHolder) return (InventoryHolder) state;
+    	return null;
+    }
+    // CraftBukkit end
 }
