--- ../src-base/minecraft/net/minecraft/item/crafting/RecipesArmorDyes.java
+++ ../src-work/minecraft/net/minecraft/item/crafting/RecipesArmorDyes.java
@@ -10,10 +10,15 @@
 import net.minecraft.item.ItemArmor;
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.World;
+import ru.svarka.crafting.ICBRecipe;
 
-public class RecipesArmorDyes implements IRecipe
+public class RecipesArmorDyes extends ShapelessRecipes implements ICBRecipe
 {
-    public boolean matches(InventoryCrafting inv, World worldIn)
+    public RecipesArmorDyes() {
+		super(new ItemStack(Items.LEATHER_HELMET, 0, 0), java.util.Arrays.asList(new ItemStack(Items.DYE, 0, 5)));
+	}
+
+	public boolean matches(InventoryCrafting inv, World worldIn)
     {
         ItemStack itemstack = null;
         List<ItemStack> list = Lists.<ItemStack>newArrayList();
