--- ../src-base/minecraft/net/minecraft/item/crafting/ShapelessRecipes.java
+++ ../src-work/minecraft/net/minecraft/item/crafting/ShapelessRecipes.java
@@ -6,8 +6,16 @@
 import net.minecraft.inventory.InventoryCrafting;
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.World;
+import ru.svarka.crafting.ICBRecipe;
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
+import org.bukkit.craftbukkit.inventory.CraftShapelessRecipe;
 
-public class ShapelessRecipes implements IRecipe
+//CraftBukkit start
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
+import org.bukkit.craftbukkit.inventory.CraftShapelessRecipe;
+// CraftBukkit end
+
+public class ShapelessRecipes implements ICBRecipe
 {
     private final ItemStack recipeOutput;
     public final List<ItemStack> recipeItems;
@@ -17,7 +25,19 @@
         this.recipeOutput = output;
         this.recipeItems = inputList;
     }
-
+    // CraftBukkit start
+        @SuppressWarnings("unchecked")
+        public org.bukkit.inventory.ShapelessRecipe toBukkitRecipe() {
+            CraftItemStack result = CraftItemStack.asCraftMirror(this.recipeOutput);
+            CraftShapelessRecipe recipe = new CraftShapelessRecipe(result, this);
+            for (ItemStack stack : (List<ItemStack>) this.recipeItems) {
+                if (stack != null) {
+                    recipe.addIngredient(org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(stack.getItem()), stack.getMetadata());
+                }
+            }
+            return recipe;
+        }
+        // CraftBukkit end
     @Nullable
     public ItemStack getRecipeOutput()
     {
